app/cards/[id]/charts/page.tsx

```diff
@@ -1,48 +1,101 @@
 "use client"
+import { differenceInMonths, parseISO } from "date-fns"
 import { useParams, useRouter } from "next/navigation"
 
 import { Chart } from "@/components/chart"
 import { Button } from "@/components/uikit/button"
 import { useCard } from "@/domains/card/hooks"
 import { useCardRecords } from "@/domains/card/record/hooks"
-import { type CardRecord } from "@/domains/card/record/types"
+import { useStaticCSV } from "@/hooks/csv"
+import { type HeightForAgeCSVData, type WeightForAgeCSVData } from "@/types/csv"
 
-import {
-  HeightAndWeightForAgeMinus2,
-  HeightAndWeightForAgePlus3,
-  WeightForHeightMinus2,
-  WeightForHeightPlus2,
-} from "./constant"
+import { WeightForHeightMinus2, WeightForHeightPlus2 } from "./constant"
 
 export default function ChartsPage() {
   const { id: idParam } = useParams()
   const cardId = idParam as string
 
   const { card } = useCard(cardId)
   const { records } = useCardRecords(cardId)
+  const weightForAgeData = useStaticCSV<WeightForAgeCSVData>(
+    "/csv/zscore/zwtageinf.csv",
+    {
+      header: true,
+      transform: (value) => Number(value),
+    },
+  )
+
+  const heightForAgeData = useStaticCSV<HeightForAgeCSVData>(
+    "/csv/zscore/zlenageinf.csv",
+    {
+      header: true,
+      transform: (value) => Number(value),
+    },
+  )
 
   const router = useRouter()
 
   if (!card) {
     return
   }
 
-  const chartData: CardRecord[] = records
+  if (!weightForAgeData || !heightForAgeData) {
+    return
+  }
+
+  const weightForAgeDataPlus2 = weightForAgeData.map((data) => {
+    return {
+      date: data.Agemos,
+      weight: data[2],
+    }
+  })
+
+  const weightForAgeDataMinus2 = weightForAgeData.map((data) => {
+    return {
+      date: data.Agemos,
+      weight: data[-2],
+    }
+  })
+
+  const heightForAgeDataPlus2 = heightForAgeData.map((data) => {
+    return {
+      date: data.Agemos,
+      height: data[2],
+    }
+  })
+
+  const heightForAgeDataMinus2 = heightForAgeData.map((data) => {
+    return {
+      date: data.Agemos,
+      height: data[-2],
+    }
+  })
+
+  const recordsWithMonths = records.map((record) => {
+    const dateOfBirth = parseISO(card.dateOfBirth)
+    const dateOfMeasurement = new Date(record.date)
+    const months = differenceInMonths(dateOfMeasurement, dateOfBirth)
```

Comment:
To remove all this converting, let's add some code in services:

`domains/card/record/services.ts`
```tsx
export function documentSnapshotToCardRecord(snapshot: DocumentSnapshot) {
  const data = snapshot.data()
  return {
    ...snapshot.data(),
    id: snapshot.id,
    date: data?.date ? new Date(data.date as string) : null,
  } as CardRecord
}
```

`domains/card/types.ts`
```tsx
export type Card = {
  id: string
  name: string
  dateOfBirth: Date
  gender: string
}
```

`domains/card/services.ts`
```tsx
export function documentSnapshotToCard(snapshot: DocumentSnapshot) {
  const data = snapshot.data()
  return {
    ...data,
    dateOfBirth: data?.dateOfBirth
      ? new Date(data.dateOfBirth as string)
      : null,
    id: snapshot.id,
  } as Card
}
```